Distinguish nightly builds with title

oguzeroglu opened the pull request:
This PR adds an indicator to Camunda Modeler titles for nightly builds.

Related to: #1419

nikku commented:
https://github.com/camunda/camunda-modeler/pull/1642/commits/2fb810da857f1fa0c7c81228e3842f56922ce4a4 is probably not the right solution to our problem.

In order to separate both apps (including cached data) we must call `app.setPath` with a different path.

nikku commented:
An alternative is to give the nightly app a different (distinct) name, i.e. call it Camunda Modeler Nightly (`camunda-modeler-nightly`) during build.

Then, once a user opens the nightly app, it is treated as a different one.

This is just one of the ideas. I'd be fine with merging the visual cue in the title only for now, too.

oguzeroglu commented:
> [2fb810d](https://github.com/camunda/camunda-modeler/commit/2fb810da857f1fa0c7c81228e3842f56922ce4a4) is probably not the right solution to our problem.
> 
> In order to separate both apps (including cached data) we must call `app.setPath` with a different path.

@nikku Thanks, I like your solution so I updated my commit.

I still allow users to define their custom user paths with user-data-dir flag.

oguzeroglu commented on app/lib/index.js:
```diff
@@ -334,6 +335,8 @@ app.createEditorWindow = function() {
   if (process.env.NODE_ENV === 'development') {
     windowOptions.title += ' (dev)';
+  } else if (app.metadata.version.includes('nightly')) {
+    windowOptions.title += ' (' + app.metadata.version.split('-')[1] + ')';
```
Yes actually setting the title in app/lib/index does nothing. I was thinking about getting rid of this during the spring cleaning, but I guess this PR is a good point to get rid of this.

barmac commented:
According to [the docs](https://electronjs.org/docs/api/browser-window), the title set in the main process has no effect if the title is set in the loaded URL. Indeed, we do that first in the [`index.html`](https://github.com/camunda/camunda-modeler/blob/develop/client/public/index.html#L10) and then programmatically (snippet commented by @nikku). I'd rather remove the two latter and leave the main process title as the only source of truth.

nikku commented:
> According to the docs, the title set in the main process has no effect if the title is set in the loaded URL.

Let's try that out -> set title in main process only.

nikku commented:
@oguzeroglu Let's also focus on the minimum and _not seperate userData paths_ for nightly and stable. This needs further consideration, documentation (plugin search paths) and so on. Let's tackle that issue at some point in the future.

oguzeroglu commented:
@nikku @barmac Ported title setting logic to backend, removed the <title> HTML tag and also removed the userPath related commit.

I think the solution is clear like this.

nikku commented:
Works. Thanks.
